pkg/apply/task/prune_task.go

```diff
@@ -50,12 +49,16 @@ func (p *PruneTask) Identifiers() []object.ObjMetadata {
 // to signal to the taskrunner that the task has completed (or failed).
 func (p *PruneTask) Start(taskContext *taskrunner.TaskContext) {
 	go func() {
-		currentUIDs := taskContext.AllResourceUIDs()
-		err := p.PruneOptions.Prune(p.InventoryObject, p.Objects,
-			currentUIDs, taskContext, prune.Options{
+		// Create filter to prevent deletion of currently applied
+		// objects. Must be done here to wait for applied UIDs.
+		uidFilter := filter.CurrentUIDFilter{
+			CurrentUIDs: taskContext.AllResourceUIDs(),
```

Comment:
Nit: We should consider renaming `AllResourceUIDs`. As far as I can tell, it actually only returns the UIDs of the resources that have been successfully applied.

Reply:
Changed `AllResourceUIDs` to `AppliedResourceUIDs`.